### PR TITLE
PI-10146 Added option to select an initial configuration

### PR DESCRIPTION
**Pull request by fedesassone**
[Jira ticket: PI-10146](https://appdirect.jira.com/browse/PI-10146)

Added an option for selecting an initial configuration with Spring, Mysql, Docker and Flyway dependencies altogether. 

screen shot 2017-07-13 at 3 58 56 pm

### CODE REVIEW

File: generators/java/templates/base/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 	<description>${project.name} Connector</description>
 
 	<properties>
-		<service-integration-sdk.version><%= sdkVersion %>
+		<service-integration-sdk.version>1.20
```
Comment (davidchau) [inline generators/java/templates/base/pom.xml]: why did you change this?

File: generators/java/templates/base/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# ${artifactId}
```
Comment (davidchau) [inline generators/java/templates/base/README.md]: why did you change this?

File: generators/java/index.js [hunk 3]
```diff
@@ -39,35 +39,64 @@ module.exports = class extends Generator {
       type: 'input',
       name: 'artifactId',
       message: 'Define your project artifactId'
-    }];
+    }, {
+      type: 'confirm',
+      name: 'diVersion',
```
Comment (davidchau) [inline generators/java/index.js]: what does diVersion mean?

File: generators/java/templates/docker/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# ${artifactId}
```
Comment (davidchau) [inline generators/java/templates/docker/README.md]: should be `<%= artifactId %>` I think

File: generators/java/templates/mysqlAndFlyway/pom.xml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>1.5.3.RELEASE</version>
+	</parent>
+
+	<description>${project.name} Connector</description>
+
+	<properties>
+		<flyway-core.version>4.0.3</flyway-core.version>
+		<mysql-connector-java.version>5.1.34</mysql-connector-java.version>
+		<service-integration-sdk.version>1.20
```
Comment (davidchau) [inline generators/java/templates/mysqlAndFlyway/pom.xml]: why are this version hardcoded? The generator is getting the version from github to get the latest one

File: generators/java/index.js [hunk 6]
```diff
@@ -39,34 +39,93 @@ module.exports = class extends Generator {
       type: 'input',
       name: 'artifactId',
       message: 'Define your project artifactId'
+    }, {
+      type: 'confirm',
+      name: 'docker',
+      message: 'Would you like to use Docker?',
+      default: false
+    },  {
```
Comment (davidchau) [inline generators/java/index.js]: space

**Comment by davidchau** [comment]
@fedesassone is this PR still valid?

**Comment by fedesassone** [comment]
@davidchau Not anymore. Thanks for pointing this up.
